packages/phoenix-otel/src/phoenix/otel/otel.py

```diff
@@ -0,0 +1,166 @@
+import inspect
+import warnings
+from typing import Any, Optional
+from urllib.parse import ParseResult, urlparse
+
+from openinference.semconv.resource import ResourceAttributes as _ResourceAttributes
+from opentelemetry import trace as trace_api
+from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import (
+    OTLPSpanExporter as _GRPCSpanExporter,
+)
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import (
+    OTLPSpanExporter as _HTTPSpanExporter,
+)
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.trace import SpanProcessor
+from opentelemetry.sdk.trace import TracerProvider as _TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor as _BatchSpanProcessor
+from opentelemetry.sdk.trace.export import SimpleSpanProcessor as _SimpleSpanProcessor
+from opentelemetry.sdk.trace.export import SpanExporter
+
+from .settings import get_env_collector_endpoint, get_env_project_name
+
+PROJECT_NAME = _ResourceAttributes.PROJECT_NAME
+
+
+def register(
+    endpoint: Optional[str] = None, project_name: Optional[str] = None, batch: bool = False
+) -> _TracerProvider:
+    """
+    Globally sets an OpenTelemetry TracerProvider for enabling OpenInference tracing.
+
+    For futher configuration, the `phoenix.otel` module provides drop-in replacements for
+    OpenTelemetry TracerProvider, SimpleSpanProcessor, BatchSpanProcessor, HTTPSpanExporter, and
+    GRPCSpanExporter objects with Phoenix-aware defaults. Documentation on how to configure tracing
+    can be found at https://opentelemetry.io/docs/specs/otel/trace/sdk/.
+
+    Args:
+        endpoint: The collector endpoint to which spans will be exported. If not provided, the
+            `PHOENIX_OTEL_COLLECTOR_ENDPOINT` environment variable will be used. The export
+            protocol will be inferred from the endpoint.
+        project_name: The name of the project to which spans will be associated. If not provided,
+            the `PHOENIX_PROJECT_NAME` environment variable will be used.
+        batch: If True, spans will be processed using a BatchSpanprocessor. If False, spans will be
+            processed one at a time using a SimpleSpanProcessor.
+    """
+
+    project_name = project_name or get_env_project_name()
+    resource = Resource.create({PROJECT_NAME: project_name})
+    tracer_provider = _TracerProvider(resource=resource)
+    span_processor: SpanProcessor
+    if batch:
+        span_processor = BatchSpanProcessor(endpoint=endpoint)
+    else:
+        span_processor = SimpleSpanProcessor(endpoint=endpoint)
+    tracer_provider.add_span_processor(span_processor)
+    trace_api.set_tracer_provider(tracer_provider)
+    return tracer_provider
+
+
+class TracerProvider(_TracerProvider):
```

Comment:
Is it really necessary to sub-class everything from OTEL and re-export them as something we would need to be maintaining going forward?